frame/democracy/src/lib.rs

```diff
@@ -0,0 +1,2084 @@
+//!   to a specific referendum.
+//!
+//! #### ExternalOrigin
+//!
+//! This call can only be made by the `ExternalOrigin`.
+//!
+//! - `external_propose` - Schedules a proposal to become a referendum once it is is legal for an
+//!   externally proposed referendum.
+//!
+//! #### External Majority Origin
+//!
+//! This call can only be made by the `ExternalMajorityOrigin`.
+//!
+//! - `external_propose_majority` - Schedules a proposal to become a majority-carries referendum
+//!   once it is legal for an externally proposed referendum.
+//!
+//! #### External Default Origin
+//!
+//! This call can only be made by the `ExternalDefaultOrigin`.
+//!
+//! - `external_propose_default` - Schedules a proposal to become a negative-turnout-bias referendum
+//!   once it is legal for an externally proposed referendum.
+//!
+//! #### Fast Track Origin
+//!
+//! This call can only be made by the `FastTrackOrigin`.
+//!
+//! - `fast_track` - Schedules the current externally proposed proposal that is "majority-carries"
+//!   to become a referendum immediately.
+//!
+//! #### Veto Origin
+//!
+//! This call can only be made by the `VetoOrigin`.
+//!
+//! - `veto_external` - Vetoes and blacklists the external proposal hash.
+//!
+//! #### Root
+//!
+//! - `cancel_referendum` - Removes a referendum.
+//! - `cancel_queued` - Cancels a proposal that is queued for enactment.
+//! - `clear_public_proposal` - Removes all public proposals.
+
+#![recursion_limit = "256"]
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::type_complexity)]
+
+use codec::{Codec, Decode, Encode, FullCodec, Input};
+use frame_support::{
+	ensure,
+	traits::{
+		schedule::{DispatchTime, Named as ScheduleNamed},
+		tokens::fungible::{
+			Inspect as NativeInspect, MutateHold as NativeMutateHold, Transfer as NativeTransfer,
+		},
+		Get, LockIdentifier,
+	},
+	transactional,
+	weights::Weight,
+};
+use orml_traits::{MultiCurrency, MultiLockableCurrency, MultiReservableCurrency};
+use scale_info::TypeInfo;
+use sp_runtime::{
+	traits::{Bounded, Dispatchable, Hash, Saturating, Zero},
+	ArithmeticError, DispatchError, DispatchResult, RuntimeDebug,
+};
+use sp_std::prelude::*;
+
+mod conviction;
+mod types;
+mod vote;
+mod vote_threshold;
+
+#[allow(clippy::all)]
+pub mod weights;
+
+pub use conviction::Conviction;
+pub use pallet::*;
+pub use types::{
+	Delegations, OriginMap, ProposalId, ReferendumInfo, ReferendumStatus, Tally, UnvoteScope,
+};
+pub use vote::{AccountVote, Vote, Voting};
+pub use vote_threshold::{Approved, VoteThreshold};
+pub use weights::WeightInfo;
+
+#[cfg(test)]
+mod tests;
+
+const DEMOCRACY_ID: LockIdentifier = *b"democrac";
+
+/// The maximum number of vetoers on a single proposal used to compute Weight.
+///
+/// NOTE: This is not enforced by any logic.
+pub const MAX_VETOERS: u32 = 100;
+
+/// A proposal index.
+pub type PropIndex = u32;
+
+/// A referendum index.
```

Comment:
seems can define trait which inherits traits and than impl if for T which has these traits and have alias like in here, but future compatible (and nightly has features for that)

Reply:
so code will not change a lot

Reply:
seems like send PR to original code to make it more typed like:)